core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTags.java

```diff
@@ -59,6 +59,21 @@ public String getKind() {
         public String getContents() {
             return contents;
         }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == null || !(other instanceof NamedStringTag)) {
+                return false;
+            }
+
+            return kind != null && kind.equals(((NamedStringTag)other).kind)
```

Comment:
Just realised this is in test code, so I really don't care that much! But comments are still worth taking on board for next time!